zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -324,34 +448,62 @@ private[codec] object EncoderDecoder                   {
     private def encodeBody(inputs: Array[Any]): Body = {
       if (isByteStream(flattened.content)) {
         Body.fromStream(inputs(0).asInstanceOf[ZStream[Any, Nothing, Byte]])
-      } else if (jsonEncoders.length == 0) Body.empty
-      else if (jsonEncoders.length == 1) {
-        val encoder = jsonEncoders(0)
+      } else {
+        if (inputs.length > 1) {
+          Body.fromMultipartForm(encodeMultipartFormData(inputs), formBoundary)
+        } else {
+          if (jsonEncoders.length == 0) Body.empty
+          else if (jsonEncoders.length == 1) {
+            val encoder = jsonEncoders(0)
+
+            encoder(inputs(0))
+          } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
```

Comment:
This branch should be unreachable as far as I can see, so maybe the tests should be against `inputs.length` and not `jsonEncoders.length`, so we don't have this unreachable `else`. Or am I missing something?